app/controllers/admin/facilities_controller.rb

```diff
@@ -118,6 +119,20 @@ def set_facility_group
     @facility_group = current_admin.accessible_facility_groups(:manage).friendly.find(params[:facility_group_id])
   end
 
+  def set_available_zones
+    zones =
+      if facility_region
+        facility_region.district.blocks
+      else
+        Region.where(type: RegionType.find_by_name("Block"))
```

Comment:
When we don't have a specific facility, we still have the context of the facility group available (see method above). We should limit the selectable blocks to one of the blocks in the facility group (district).